src/http/request.cr

```diff
@@ -246,14 +246,40 @@ class HTTP::Request
   end
 
   # Returns request host from headers.
+  @[Deprecated("Use `#hostname` instead.")]
   def host
     host = @headers["Host"]?
     return unless host
     index = host.index(":")
     index ? host[0...index] : host
   end
 
+  # Extracts the hostname from `Host` header.
+  #
+  # Returns `nil` if the `Host` header is missing.
+  #
+  # If the `Host` header contains a port number, it is stripped off.
+  def hostname : String?
+    header = @headers["Host"]?
+    return unless header
+
+    host, _, port = header.rpartition(":")
+    if host.empty?
+      # no colon in header
+      host = header
+    else
+      port = port.to_i?(whitespace: false)
+      unless port && {% if flag?(:win32) %}port.in?(0..UInt16::MAX){% else %}Socket::IPAddress.valid_port?(port){% end %}
```

Comment:
Why not use `Socket::IPAddress.valid_port?` in `win32` too?

```suggestion
      unless port && Socket::IPAddress.valid_port?(port)
```

Reply:
`Socket` hasn't been ported yet. So this is a temporal fix.

Reply:
I'd leave a `TODO` / `FIXME` note to keep it in mind for the future.

Reply:
Yeah, added